### PR TITLE
Start and open workspace

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Adds project stop and fullscreen icons
- Adds spinner animation
- Adds `Ide` component as a container for the IDE

### Important Notes
This **does not work** on the desktop IDE as it runs `enso.main`

### Screencast
https://user-images.githubusercontent.com/4046547/224118329-9107e855-b517-4dc0-a445-7195662206ed.mp4

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Comment by somebody1234** [comment]
Not working after merging so I'm keeping it as a draft

**Comment by somebody1234** [comment]
Seems to work now, I didn't change anything though - backend might just have been having issues yesterday

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx [hunk 1]
```diff
@@ -13,49 +13,41 @@ import withRouter from '../../navigation'
...
+import * as useProjectManagerAdapter from '../useProjectManagerAdapter'
 
 // ==========================
 // === dashboardContainer ===
 // ==========================
 
+enum Tab {
+  dashboard = "dashboard",
+  ide = "ide",
+}
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx]: It's not obvious what this struct does by just looking at the code. In such cases, please add a shot doc comment with explanation of what is it.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/** @file  */
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: The linter tells about missing doc comment, but leaving it in this way is not the proper way of documenting the codebase. We care a LOT about proper documentation. Please write here good explanation of what this file does and how it semantically works.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import * as react from "react";
+
+import * as service from "../service";
+
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: This file does not contain code sections. See our code style guide - it contains explanation where to put sections, please.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+const IDE_CDN_URL = "https://ensocdn.s3.us-west-1.amazonaws.com/ide";
+
+interface Props {
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx:16]: We are trying not to use abbreviations in class and functions names. Abbreviations are ok in local variables in Fns, but lets keep the names full otherwise, please.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  project: service.Project;
+  backendService: service.Backend;
+}
+
+declare global {
+  const enso: { main: (config: object) => void; };
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: Why do we need to disable the lint?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+const Ide = ({ project, backendService }: Props) => {
+  react.useEffect(() => {
+    void (async () => {
+      const ideVersions = await backendService.listVersions({
+        versionType: service.VersionType.ide,
+        default: true,
+      });
+      /* TODO: make latest nightlies configurable again */
+      /* TODO: reload style.css and index.js.gz when opening a different version */
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: todo left in the code: Do one of the following points:
1. Fix it in this PR.
2. Or, chat with your Project Leader and agree that it will be fixed as a separate PR. In this case, an issue should be created, described and linked in the TODO comment. You should also annotate the TODO comment with you initials. Again, see code style guide to learn more.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      const projectIdeVersion = "2023.1.1-nightly.2023.2.9"; // project.ideVersion?.number.value ?? ideVersions[0]!.number.value;
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: 1. hardcoded value
2. commented code

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        versionType: service.VersionType.ide,
+        default: true,
+      });
+      const backendVersions = await backendService.listVersions({
+        versionType: service.VersionType.backend,
+        default: true,
+      });
+      console.log('no...', project);
+      const projectIdeVersion = "2023.1.1-nightly.2023.2.9"; // project.ideVersion?.number.value ?? ideVersions[0]!.number.value;
+      const projectEngineVersion = "2023.1.1-nightly.2023.2.9"; // project.engineVersion?.number.value ?? backendVersions[0]!.number.value;
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: a lot of hardcoded values here

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      enso.main({
+        /* eslint-disable @typescript-eslint/naming-convention */
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: why the lint needs to be disabled?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        languageServerRpc: `${project.address!}json`,
+        languageServerData: `${project.address!}binary`,
+        isInCloud: true,
+        useLoader: true,
+        authenticationEnabled: false,
+        project: project.packageName,
+        preferredEngineVersion: projectEngineVersion,
+        dataGathering: true,
+        email: user!.userEmail,
+        /* eslint-enable @typescript-eslint/naming-convention */
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: why the lint needs to be disabled?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
+/** @file A label that may have a warning symbol */
+import * as react from "react";
+
+export enum Status {
+	none = "none",
+	warning = "warning",
+	severeWarning = "severe-warning",
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx]: strange indentation - did you seta's instead of spaces? Please be sure that everywhere you are using spaces and the indentation is consistent.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+const ICONS = {
+	none: <></>,
+	exclamation: <svg width="27" height="27" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
+		<path fill="#F9FAFB" fillOpacity="0.72" fillRule="evenodd" d="M9 0A9 9 0 1 1 9 18 9 9 0 1 1 9 0M7.5 3.5H10.5L10 10.5H8L7.5 3.5ZM8 12L10 12 10 14 8 14"/>
+	</svg>,
+};
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx]: no documentation about what this is.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+	[Status.none]: ICONS.none,
+	[Status.warning]: ICONS.exclamation,
+	[Status.severeWarning]: ICONS.exclamation,
+};
+
+interface Props {
+	status?: Status;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx]: why the lint needs to be disabled?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+const Label = ({ status = Status.none, children }: react.PropsWithChildren<Props>) => {
+	return <div className={`status-${status} whitespace-nowrap rounded-full inline-flex flex-row flex-nowrap align-middle leading-[27px] p-1 m-1`}>{statusIcon[status]}<div className="px-1">{children}</div></div>;
+}
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/label.tsx]: Again, short documentation describing what this is would be nice to have here. Please add it here and in other similar places.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/permissionDisplay.tsx [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  return <div className="m-1 relative inline-block">
+    {permissionBorder}
+    <div className="bg-label-bg rounded-full px-4 py-1 m-1">{children}</div>
+  </div>;
+}
+
+
+export default PermissionDisplay;
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/permissionDisplay.tsx]: no nl at the end of the file. Please be sure that you formatted all source code with prettier.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx [hunk 16]
```diff
@@ -67,69 +73,43 @@ const ProjectActionButton = (props: Props) => {
             return (
                 <>
                     <button onClick={handleOpenProject}>
-                        <heroicons.PlayCircleIcon className="h-10 w-10 text-grey-500" />
+                        <svg width={36} height={36} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+                            <path d="m10.04 7.34 6 3.85a1 1 0 0 1 0 1.68l-6 3.85a1 1 0 0 1-1.54-.84v-7.7a1 1 0 0 1 1.54-.84Z" fill="currentColor" fillOpacity=".8"/>
+                            <rect x={1.5} y={1.5} width={21} height={21} rx={10.5} stroke="currentColor" strokeOpacity={0.1} strokeWidth={3}/>
+                        </svg>
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx]: this is magic. Such things should be exported to variables with nice names of what they define

**Review by wdanilo — CHANGES_REQUESTED** [review]
1. I left a few comments. No worries, first PR is always like that. Please follow our [code style guide](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md).

2. Also, every PR should have screenshots or screencasts showing how it works.
3. The PR does not have the checkboxes checked. Please be sure you check them before requesting a review :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 5 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: currently the `naming-convention` lint is explicitly disabled with a comment without further explanation for every react component, since react components use PascalCase which isn't allowed by `naming-convention` by default

Comment (wdanilo) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: Let's fix the linter config then :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 7 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: intentional, as of right now the latest nightly doesn't accept parameters in `enso.main` - not sure what the correct solution would be. the commented code is the actual code that should be there, but it currently doesn't work

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+      const user = await backendService.getUser();
+      const ideVersions = await backendService.listVersions({
+        versionType: service.VersionType.ide,
+        default: true,
+      });
+      const backendVersions = await backendService.listVersions({
+        versionType: service.VersionType.backend,
+        default: true,
+      });
+      console.log('no...', project);
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: debug logging left in, will fix asap

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 9 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: the config object was using `snake_case` because the development environment was using an outdated engine version, forgot to remove it when switching back to `camelCase`

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 4 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx:16]: if you mean `interface Props`, i believe the props interfaces for all other components are also called `Props` - should those all be changed too?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 5 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: I feel like that would belong in a separate PR? since it would involve changing everywhere else that has the naming-convention lint disabled

Comment (wdanilo) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: In a separate PR we can fix the existing places, but disabling lints in a new code is always a no-go for us if this is not really necessary / this is not the correct solution. Unfortunately, there exist a code with bad things. It was probably not caught during previous PR reviews. Anyway, I believe that the new code should not have it if we can fix it simply. I understand that by fixing the linter config you do not need to modify the old code, right?

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: right

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/permissionDisplay.tsx (hunk 15 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/permissionDisplay.tsx]: much of the code seems to not have been formatted with prettier, i'm assuming the same applies here - format all files changed in this PR, and have a separate PR that formats all other files

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/permissionDisplay.tsx]: *never mind - maybe the prettier config was messed up before, most of the code seems to be fine now

**Comment by somebody1234** [comment]
:warning: this PR is not ready for merge, waiting on the backend's default version to be updated to at least `2023.1.1-nightly.2023.3.8.1` so that initializing the IDE (`enso.main()`) actually opens up the project

**Comment by somebody1234** [comment]
this force-push is just to change the environment back to production

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+            const backendVersion = (
+                await backendService.listVersions({
+                    versionType: service.VersionType.backend,
+                    default: true,
+                })
+            )[0]
+            const projectIdeVersion = project.ideVersion?.value ?? ideVersion.number.value
+            const projectEngineVersion = project.engineVersion?.value ?? backendVersion.number.value
+            console.log('project', project)
+            void window.enso.main({
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: missing `await`

**Review by PabloBuchu — CHANGES_REQUESTED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 18 above)
Comment (PabloBuchu) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: also fron time to time I get probably some race condition that `window.enso` is not defined

Screenshot 2023-03-30 at 14 22 57

**Comment by PabloBuchu** [comment]
@Nctdt can you intergrate here your icon `my current project` from https://github.com/enso-org/enso/pull/6064 and connect with logic? So we can switch between currently opened project and dashboard

**Comment by PabloBuchu** [comment]
@somebody1234 could you take over moving the current open project icon here from the PR https://github.com/enso-org/enso/pull/6064?

**Comment by somebody1234** [comment]
@PabloBuchu do you mean merging #6064 into here?

**Comment by somebody1234** [comment]
(github web ui said it couldn't merge again, but git cli merged it just fine again)

**Comment by PabloBuchu** [comment]
@somebody1234 three things:
1. When openning a project a loader covers the whole screen. Can we leave the top bar on top?
2. This also should be in the size of window. Currently its fixed and plus icon is partially hidden and scroll bar is forced
Screenshot 2023-04-11 at 13 01 41

3. We have css leak:
Screenshot 2023-04-11 at 13 02 46

**Comment by somebody1234** [comment]
1. yup, but i think i'll need to make things work on the desktop IDE first? since the cloud dashboard pulls from the backend's version so local edits will never show up. it's probably *theoretically* possible to get cloud dashboard to work with the local backend, i could try if you think it's better to fix this before fixing authentication on desktop
2. yeah, same issue as above
3. :ok_hand: will investigate... sometime

**Comment by somebody1234** [comment]
*not* ready.
to do - i'll do these later:
- confirm that i didn't mess anything up in `dashboard.tsx`
- hide top bar
- investigate & fix css issue

**Comment by PabloBuchu** [comment]
@somebody1234 let me know if we can add here handling opening project by project manager here if running in electron or we need separate PR for that

**Comment by somebody1234** [comment]
@PabloBuchu i think enough changes are needed that it *should* be a separate PR - it will take quite a while to do and will likely be quite a significant change to lots of files too

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx (hunk 18 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/ide.tsx]: this race condition *should* be solved; `void` -> `await` has also been fixed

**Comment by somebody1234** [comment]
no changelog required, because if one is required it should be only done once, either on the first one (#6015) or the last one (not yet opened)

**Comment by somebody1234** [comment]
@PabloBuchu just a heads up - the top bar is now being hidden on IDE view, however that means the animation is no longer visible, because the tab is immediately switched to the IDE tab. it's probably possible to let the animation play, but not sure whether it's worth the added complexity for the time being

**Comment by somebody1234** [comment]
importantly however, this means there will be no way to get back to the dashboard (other than manually refreshing) until the IDE adds the event that we can listen to - since the switcher will no longer be visible.

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
FYI This PR only adds support for openning projects from a cloud. We will add integration with project manager in the upcoming one not to touch to many files at once

**Review by wdanilo — APPROVED** [review]

**Comment by PabloBuchu** [comment]
@somebody1234 please fix merge conflicts so the PR can be merged
